### PR TITLE
Updated usaco-966.mdx with Java Code

### PR DESCRIPTION
**Pull request by khan7SAK**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-966.mdx [hunk 1]
```diff
@@ -47,4 +47,46 @@ ans += first8[N % 8]
...
+
+
+class Usaco
+{
+	public static void main (String[] args) throws java.lang.Exception
+	{
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int N = Integer.parseInt(st.nextToken());
+		
+		Vector<Integer> v = new Vector<Integer>(N);
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:66]: Why not `ArrayList`?

File: solutions/usaco-966.mdx [hunk 2]
```diff
@@ -47,4 +47,46 @@ ans += first8[N % 8]
...
+		v.add(2);
+		v.add(4);
+		v.add(7);
+		v.add(8);
+		v.add(11);
+		v.add(13);  
+		 
+		int res = 0;
+	// Finds the group
+	res = (n - 1) / 8;
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:78]: indentation weird, u need another tab here and after

File: solutions/usaco-966.mdx [hunk 3]
```diff
@@ -47,4 +47,46 @@ ans += first8[N % 8]
 print(ans)
 ```
 </PySection>
+<JavaSection>
+  
+```java
+
+import java.util.*;
+import java.lang.*;
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:55]:
```suggestion

```
do we really need it?

File: solutions/usaco-966.mdx [hunk 4]
```diff
@@ -47,4 +47,46 @@ ans += first8[N % 8]
...
+import java.io.*;
+
+
+class Usaco
+{
+	public static void main (String[] args) throws java.lang.Exception
+	{
+		StringTokenizer st = new StringTokenizer(br.readLine());
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:63]:
```suggestion
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
```

where does ur `br` come from? Forgot a line `BufferedReader br = new BufferedReader(new InputStreamReader(System.in))` before? 
And just throw `IOException`

File: solutions/usaco-966.mdx [hunk 5]
```diff
@@ -47,4 +47,46 @@ ans += first8[N % 8]
...
+		v.add(13);  
+		 
+		int res = 0;
+	// Finds the group
+	res = (n - 1) / 8;
+	
+	// Group every 15 numbers
+	res *= 15;
+	
+	//responding leftover
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:83]:
```suggestion
	// Responding leftover
```
and u mean *cor*responding or responding?

**Comment by ChuangSheep** [comment]
Also, maybe consider using the "conventional" bracket style, for a reference u can see https://usaco.guide/problems/usaco-736-bovine-genomics/solution
and maybe rename the class to the problem name

File: solutions/usaco-966.mdx (hunk 1 above)
Comment (khan7SAK) [reply solutions/usaco-966.mdx:66]: It can done using both ArrayList and Vector.

File: solutions/usaco-966.mdx (hunk 4 above)
Comment (khan7SAK) [reply solutions/usaco-966.mdx:63]: I'm sorry ,while submitting the code I missed "BufferedReader br = new BufferedReader(new InputStreamReader(System.in))".I did test the code with good number of Inputs.

File: solutions/usaco-966.mdx (hunk 3 above)
Comment (khan7SAK) [reply solutions/usaco-966.mdx:55]: No, java. lang package is a default package in Java. Hence, we don't really need to import it explicitly.

File: solutions/usaco-966.mdx [hunk 6]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
 </PySection>
+<JavaSection>
+  
+```java
+
+import java.util.*;
+import java.io.*;
+
+
+class usaco-966
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:58]:
```suggestion
public class MooBuzz
```
With "Problem name" I meant `public class MooBuzz`, Java won't compile with slash in class name.
Ah and if class isn't public it won't work anyway

File: solutions/usaco-966.mdx [hunk 7]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+{
+	public static void main (String[] args) throws IOException
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:60]:
```suggestion
	public static void main(String[] args) throws IOException
```
spacing

File: solutions/usaco-966.mdx [hunk 8]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+	{
+
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:62]:
```suggestion
```
spacing

File: solutions/usaco-966.mdx [hunk 9]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+		v.add(4);
+		v.add(7);
+		v.add(8);
+		v.add(11);
+		v.add(13);  
+		 
+		int res = 0;
+		
+		// Finds the group
+		res = (n - 1) / 8;
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:80]:
```suggestion
		res = (N - 1) / 8;
```
`n` undefined, doesn't compile

File: solutions/usaco-966.mdx [hunk 10]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+	
+		// Group every 15 numbers
+		res *= 15;
+	
+		//Coresponding leftover
+		res += v.get(n % 8);
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:86]:
```suggestion
		res += v.get(N % 8);
```
same for this one

File: solutions/usaco-966.mdx [hunk 11]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
 ```
 </PySection>
+<JavaSection>
+  
+```java
+
+import java.util.*;
+import java.io.*;
+
+
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:57]:
```suggestion
```
spacing

File: solutions/usaco-966.mdx [hunk 12]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+class usaco-966
+{
+	public static void main (String[] args) throws IOException
+	{
+	
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int N = Integer.parseInt(st.nextToken());
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:65]:
```suggestion
		BufferedReader br = new BufferedReader(new FileReader("moobuzz.in"));
		int N = Integer.parseInt(br.readLine());
		br.close();
```
And USACO 2019 still uses FileIO, so the IO previously wouldn't work anyway

File: solutions/usaco-966.mdx [hunk 13]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+		// Finds the group
+		res = (n - 1) / 8;
+	
+		// Group every 15 numbers
+		res *= 15;
+	
+		//Coresponding leftover
+		res += v.get(n % 8);
+                
+		System.out.println(res);
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:88]:
```suggestion
		PrintWriter pw = new PrintWriter("moobuzz.out");
		pw.println(res);
		pw.close();
```
same, u need FileIO

File: solutions/usaco-966.mdx [hunk 14]
```diff
@@ -47,4 +47,48 @@ ans += first8[N % 8]
...
+class usaco-966
+{
+	public static void main (String[] args) throws IOException
+	{
+	
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int N = Integer.parseInt(st.nextToken());
+		
+		Vector<Integer> v = new Vector<Integer>(N);
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:67]: You can't initialize your list with `N` as capacity. N can up to 10^9 which will lead to RE/memory limit exceeded. And you only want to save a few **fixed** numbers anyway. (Actually, *why not use an array?*)

**Review by ChuangSheep — COMMENTED** [review]
> I did test the code with good number of Inputs.

idk if the code doesn't really compile...

>  I have tested my code.

That means *I have submitted my code to the online Judge (e.g. USACO.org) and got AC*

Ah and feel free to click the "resolve discussion" button when you are finished with one issue since I can't...

File: solutions/usaco-966.mdx [hunk 15]
```diff
@@ -47,4 +47,41 @@ ans += first8[N % 8]
...
+
+public class MooBuzz 
+{
+	public static void main(String[] args) throws IOException
+	{
+		BufferedReader br = new BufferedReader(new FileReader("moobuzz.in"));
+		int N = Integer.parseInt(br.readLine());
+		br.close();
+		 
+		long[] v={14,1,2,4,7,8,11,13};
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:66]:
```suggestion
		long[] v = {14, 1, 2, 4, 7, 8, 11, 13};
```

File: solutions/usaco-966.mdx [hunk 16]
```diff
@@ -47,4 +47,41 @@ ans += first8[N % 8]
...
+		 
+		int res = 0;
+		
+		// Finds the group
+		res = (N - 1) / 8;
+	
+		// Group every 15 numbers
+		res *= 15;
+	
+		//Coresponding leftover
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:76]:
```suggestion
		// Coresponding leftover
```
spacing pls

File: solutions/usaco-966.mdx [hunk 17]
```diff
@@ -47,4 +47,41 @@ ans += first8[N % 8]
...
+		res += v.get(N % 8);
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:77]:
```suggestion
		res += v[N % 8];
```
array doesn't have `get`

**Review by ChuangSheep — COMMENTED** [review]
and aside from the bracket style, everyting else looks good

File: solutions/usaco-966.mdx (hunk 11 above)
Comment (ChuangSheep) [reply solutions/usaco-966.mdx:57]: emmm... this one is actually the only one not resolved yet...

Comment (khan7SAK) [reply solutions/usaco-966.mdx:57]: Done

File: solutions/usaco-966.mdx [hunk 18]
```diff
@@ -47,4 +47,38 @@ ans += first8[N % 8]
...
+		int res = 0;
+		
+		// Finds the group
+		res = (N - 1) / 8;
+	
+		// Group every 15 numbers
+		res *= 15;
+	
+		// Coresponding leftover
+		res += v[N % 8]
```
Comment (ChuangSheep) [inline solutions/usaco-966.mdx:74]:
```suggestion
		res += v[N % 8];
```

**Review by ChuangSheep — APPROVED** [review]
Nice! 🎉

File: solutions/usaco-966.mdx (hunk 18 above)
Comment (khan7SAK) [inline solutions/usaco-966.mdx]: res += v[N % 8];

**Comment by ChuangSheep** [comment]
@SansPapyrus683 👀

File: solutions/usaco-966.mdx [hunk 19]
```diff
@@ -47,4 +47,38 @@ ans += first8[N % 8]
...
+import java.util.*;
+import java.io.*;
+
+public class MooBuzz {
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new FileReader("moobuzz.in"));
+		int N = Integer.parseInt(br.readLine());
+		br.close();
+		 
+		int[] v = {14, 1, 2, 4, 7, 8, 11, 13};
```
Comment (SansPapyrus683) [inline solutions/usaco-966.mdx:63]: rename to `first8` for consistency

**Comment by SansPapyrus683** [comment]
btw there's a merge conflict, can you resolve it?

File: solutions/usaco-966.mdx [hunk 20]
```diff
@@ -79,6 +79,39 @@ int main() {
...
+                
+		PrintWriter pw = new PrintWriter("moobuzz.out");
+		pw.println(res);
+		pw.close();
+	}
+}
 
-
+```
+</JavaAection>
```
Comment (khan7SAK) [inline solutions/usaco-966.mdx]: <JavaSection>

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**